server/pulp/server/managers/repo/unit_association_query.py

```diff
@@ -115,9 +116,23 @@ def get_units(self, repo_id, criteria=None, as_generator=False):
         # Use a generator expression here to keep from going back to the types
         # collections once we've returned our limit of results.
         # Be sure to skip cursors that would otherwise return an empty result set.
-        units_cursors = (self._associated_units_by_type_cursor(t, criteria,
-                                                               associations_lookup[t].keys())
-                         for t in association_unit_types if t in associations_lookup)
+        if criteria.unit_sort:
+            units_cursors = (self._associated_units_by_type_cursor(t, criteria,
+                                                                   associations_lookup[t].keys())
+                             for t in association_unit_types if t in associations_lookup)
+        else:
```

Comment:
if you could leave here a comment with an explanation something along the lines https://pulp.plan.io/issues/2220#note-2 that would be very helpful whenever we'd look into this code in the future.

Reply:
Added comments to explain why and how to change.